### PR TITLE
Add closeClient for all cases MODUSERBL-147

### PR DESCRIPTION
**Pull request by adamdickmeiss**
Not pretty as socket is  never reused, but should avoid resource
leaks for now. Many code smells not fixed by this. Just keeping it
small.

### CODE REVIEW

File: src/main/java/org/folio/rest/impl/BLUsersAPI.java [hunk 1]
```diff
@@ -535,6 +538,7 @@ public void getBlUsers(String query, int offset, int limit,
     try {
       userIdResponse[0] = client.request(userUrl.toString(), okapiHeaders);
     } catch (Exception ex) {
+      client.closeClient();
```
Comment (julianladisch) [inline src/main/java/org/folio/rest/impl/BLUsersAPI.java:541]: All lines after `HttpClientInterface client = HttpClientFactory.getHttpClient(okapiURL, tenant);` should be included into the try clause.

File: src/main/java/org/folio/rest/impl/BLUsersAPI.java [hunk 2]
```diff
@@ -802,6 +807,7 @@ public void postBlUsersLogin(boolean expandPerms, List<String> include, String u
     }
 
     if (entity == null || entity.getUsername() == null || entity.getPassword() == null) {
+      client.closeClient();
```
Comment (julianladisch) [inline src/main/java/org/folio/rest/impl/BLUsersAPI.java:810]: The next catch clause (line 828) also needs a close.

Comment (adamdickmeiss) [reply src/main/java/org/folio/rest/impl/BLUsersAPI.java:810]: Damn. A very bad style

File: src/main/java/org/folio/rest/impl/BLUsersAPI.java (hunk 1 above)
Comment (adamdickmeiss) [reply src/main/java/org/folio/rest/impl/BLUsersAPI.java:541]: Not possible as userIdResponse is used after the block.

Comment (adamdickmeiss) [reply src/main/java/org/folio/rest/impl/BLUsersAPI.java:538]: Moved some in there, just now.

**Review by julianladisch — APPROVED** [review]
Looks good, thanks!

**Comment by adamdickmeiss** [comment]
> Looks good, thanks!

Not pretty IMHO, but hopefully it improves the sitaution a bit.

**Pull request merged**